test/test_rpc.py

```diff
@@ -77,9 +79,9 @@ def _wrap_with_rpc(func):
     '''
     def wrapper(self):
         store = dist.FileStore(self.file.name, self.world_size)
-        dist.init_process_group(backend='gloo', rank=self.rank,
-                                world_size=self.world_size, store=store)
-        dist.init_model_parallel('worker%d' % self.rank)
+        dist.init_model_parallel('worker%d' % self.rank, rpc_backend=BACKEND,
+                                 rank=self.rank, world_size=self.world_size,
+                                 store=store)
```

Comment:
Had a discussion with @mrshenli and want to document it here to solicit more feedback.

This diff changes init_model_parallel behaviour and calls init_process_group internally. Since all backends may not support pg like collectives this may make it inconvenient for the user. Also this makes the behavior different than data parallel case where init_process_group should have been initialized before DistributedDataParallel. Another argument against doing this is that each rpc_backend may need different arguments and hence we would need to keep the arguments as a supper set of the all backends.

Based on the above I plan to keep init_process_group to be called if needed by the user outside init_model_parallel and make the arguments to init_model_parallel as a url which can be interpreted by the backend implementation.